File: API/DTOs/Filtering/SortOptions.cs
namespace API.DTOs.Filtering;

/// <summary>
/// Sorting Options for a query
/// </summary>
public sealed record SortOptions
{
    public SortField SortField { get; set; }
    public bool IsAscending { get; set; } = true;
}

/// <summary>
/// All Sorting Options for a query related to Person Entity
/// </summary>
public sealed record PersonSortOptions
{
    public PersonSortField SortField { get; set; }
    public bool IsAscending { get; set; } = true;
}

File: UI/Web/src/app/_models/metadata/v2/sort-options.ts
import {PersonSortField} from "./person-sort-field";

/**
 * Series-based Sort options
 */
export interface SortOptions<TSort extends number = number> {
  sortField: TSort;
  isAscending: boolean;
}

/**
 * Person-based Sort Options
 */
export interface PersonSortOptions {
  sortField: PersonSortField;
  isAscending: boolean;
}
